src/pain/001/SEPACreditPaymentInitiation.ts

```diff
@@ -0,0 +1,148 @@
+      instruction => !instruction.creditor.address,
+    );
+
+    const creditorWithIncompleteAddress = this.paymentInstructions.find(
+      instruction => {
+        const address = instruction.creditor.address;
+        return !address || !address.country;
+      },
+    );
+
+    if (creditorWithIncompleteAddress) {
+      throw new Error(
+        'All creditors must have complete addresses (street name, building number, postal code, town name, and country)',
+      );
+    }
+  }
+
+  paymentInformation(instruction: SEPACreditPaymentInstruction) {
+    const paymentInfoID = sanitize(instruction.id || uuidv4(), 35);
+    instruction.dinero ??= Dinero({ amount: instruction.amount, currency: instruction.currency });
+
+    return {
+      PmtId: {
+        InstrId: paymentInfoID,
+        EndToEndId: paymentInfoID,
+      },
+      Amt: {
+        InstdAmt: {
+          '#': instruction.dinero.toFormat('0.00'),
+          '@Ccy': instruction.currency,
+        },
+      },
+      CdtrAgt: this.agent(instruction.creditor.agent as BICAgent),
+      Cdtr: this.party(instruction.creditor as Party),
+      CdtrAcct: {
+        Id: { IBAN: (instruction.creditor.account as IBANAccount).iban },
+        Ccy: instruction.currency,
+      },
+      RmtInf: instruction.remittanceInformation ? {
+        Ustrd: instruction.remittanceInformation,
+      } : undefined,
+    };
+  }
+
+  public serialize(): string {
+    const xml = {
+      Document: {
+        '@xmlns': 'urn:iso:std:iso:20022:tech:xsd:pain.001.001.03',
+        '@xmlns:xsi': 'http://www.w3.org/2001/XMLSchema-instance',
+        CstmrCdtTrfInitn: {
+          GrpHdr: {
+            MsgId: this.messageId,
+            CreDtTm: this.creationDate.toISOString(),
+            NbOfTxs: this.paymentInstructions.length.toString(),
+            CtrlSum: this.paymentSum,
+            InitgPty: {
+              Nm: this.initiatingParty.name,
+              Id: {
+                OrgId: {
+                  Othr: {
+                    Id: this.initiatingParty.id,
+                  },
+                },
+              },
+            },
+          },
+          PmtInf: {
+            PmtInfId: `TRF${Date.now()}`,
+            PmtMtd: 'TRF',
+            NbOfTxs: this.paymentInstructions.length.toString(),
+            CtrlSum: this.paymentSum,
+            PmtTpInf: {
+              SvcLvl: { Cd: 'SEPA' },
+              CtgyPurp: { Cd: 'TRAD' },
+            },
+            ReqdExctnDt: this.creationDate.toISOString().split('T').at(0),
+            Dbtr: this.party(this.initiatingParty),
+            DbtrAcct: this.account(this.initiatingParty.account as Account),
+            DbtrAgt: this.agent(this.initiatingParty.agent as Agent),
+            ChrgBr: 'SLEV',
+            // payments[]
+            CdtTrfTxInf: this.paymentInstructions.map(p => this.paymentInformation(p)),
+          }
+        }
+      },
+    };
+
+    const doc = create(xml);
+    return doc.end({ prettyPrint: true });
```

Comment:
I wonder if this is necessary since it might be a little wasteful. We do it for SWIFT as well, so I think this is okay for now. https://github.com/Svapnil/iso20022.js/blob/aa63c1b60433a3e22654d78b489f0ff59d205742/src/pain/001/SWIFTCreditPaymentInitiation.ts#L177

Reply:
I was thinking that this could maybe be an optional parameter, somewhere. Where would we put that, or are we interested in doing that?